jobclient/src/main/java/com/twosigma/cook/jobclient/constraint/Constraints.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) Two Sigma Open Source, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.twosigma.cook.jobclient.constraint;
+
+import com.google.common.base.Preconditions;
+import com.twosigma.cook.jobclient.constraint.api.Constraint;
+import com.twosigma.cook.jobclient.constraint.api.ConstraintBuilder;
+import com.twosigma.cook.jobclient.constraint.api.Operator;
+import org.json.JSONArray;
+
+
+/**
+ * A constraint in Cook could be one of the following three forms
+ * <ul>
+ * <li>attribute, operator
+ * <li>attribute, operator, value
+ * <li>attribute, operator, list of values
+ * </ul>
+ * For examples,
+ * <ul>
+ * <li>"host", UNIQUE
+ * <li>"host", EQUALS, foo.bar.com
+ * <li>"host", IN, [foo1.bar.com,foo2.bar.com]
+ * </ul>
+ * For now, only EQUALS operator is supported by Cook. One could construct
+ * a constraint with an "EQUALS" operator by
+ * <p>
+ * <pre>
+ *  Constraint c = Constraints.get(Operator.EQUALS).build("host", "foo.bar.com");
+ * </pre>
+ */
+public class Constraints {
+
+    /**
+     * Return a {@link ConstraintBuilder} specific for the given operator.
+     *
+     * @param operator the constraint operator
+     * @return a {@link ConstraintBuilder} specific for the given operator.
+     */
+    public static ConstraintBuilder get(Operator operator) {
+        switch (operator) {
+            case EQUALS:
+                return new OneToOneConstraintBuilder(operator);
+            default:
+                throw new UnsupportedOperationException(operator + " is not supported.");
+        }
+    }
+
+    public static Constraint parseFrom(JSONArray constraint) {
```

Comment:
Once we have the operator, we can call `operator.parseFrom(constraint)` with the necessary casting, where `parseFrom` is an abstract method on the `enum Operator` and avoid the need for `ConstraintBuilder.parseFrom`.